lib/mail.ex

```diff
@@ -78,21 +97,36 @@ defmodule Mail do
 
   If a text part already exists this function will replace that existing
   part with the new part.
+
+  ## Options
+
+  * `:charset` - The character encoding standard for content type
   """
-  def put_html(%Mail.Message{multipart: true} = message, body) do
+  def put_html(message, body, opts \\ [])
+
+  def put_html(%Mail.Message{multipart: true} = message, body, opts) do
     message =
       case Enum.find(message.parts, &Mail.Message.match_content_type?(&1, "text/html")) do
         %Mail.Message{} = part -> Mail.Message.delete_part(message, part)
         _ -> message
       end
 
-    Mail.Message.put_part(message, Mail.Message.build_html(body))
+    Mail.Message.put_part(message, Mail.Message.build_html(body, opts))
   end
 
-  def put_html(%Mail.Message{} = message, body) do
+  def put_html(%Mail.Message{} = message, body, opts) do
+    content_type =
+      case opts do
+        charset: charset ->
+          ["text/html", {"charset", charset}]
+
+        _else ->
+          "text/html"
+      end
+
     Mail.Message.put_body(message, body)
     |> Mail.Message.put_header(:content_transfer_encoding, :quoted_printable)
-    |> Mail.Message.put_content_type("text/html")
+    |> Mail.Message.put_content_type(content_type)
```

Comment:
If this is meant to be specifically a "html" part why would we want the `content_type` to be overridable?

Reply:
Oh I see below, it looks like you force the content type but allow charset to be configurable?

Reply:
Correct. Some local mail clients do not display mail properly if charset is not set. (Japanese in my case)